common/kvstore/src/main/java/org/apache/spark/util/kvstore/InMemoryStore.java

```diff
@@ -106,6 +106,12 @@ public void delete(Class<?> type, Object naturalKey) {
     return list != null ? list.view() : emptyView();
   }
 
+  @Override
+  public <T> KVStoreView<T> viewWithCondition(Class<T> type, Predicate<T> condition) {
+    InstanceList<T> list = inMemoryLists.get(type);
```

Comment:
Rather than leave the code above duplicating this, have it call this one with a null condition and handle it here. Subclasses would only override `viewWithCondition` then.

Reply:
I’ve tried like `view(Class<T> type, Optional<Predicate<T>> condition)`, but how can I add a default parameter value in Java like `condition: Option[Predicate[T]] = None` in Scala? Otherwise, it needs to change many places so I choose to add a new function called `viewWithCondition`.

Reply:
You can't create a default parameter value, but you can keep both methods. I'm just suggesting there's no need to duplicate their logic here and in subclasses.

Reply:
Oh, get it

Reply:
Dedup like this causes thing worse since `view()` would traverse all items. It was just return `map.values()`

Reply:
You're still going to exam all items, right? I don't see where this can be pushed down enough to make it faster.

Reply:
There are low level APIs, the different between their efficiencies doesn’t depend on the API self, it depends on the invocation layer. For example the logic of invocation layer uses `view` to get all items and creates some kinds of objects for them, then it executes multiple heavy logic, finally it uses a filter to get less result. So using `viewWithCondiction` is very like SQL prediction pushdown optimization rule. Even the cost of the filter operation itself always fixed. But filter out as earlier as possible still could save memory and time.